### PR TITLE
feat(datapoints-graph): alarm event selector list for Data points graph 2.0

### PR DESCRIPTION
**Pull request by jdreSAG**
Alarm and event selector module and list component.
Implementation for displaying alarms and events on graph - https://github.com/SoftwareAG/cumulocity-community-plugins/pull/28
Integration branch for selector and graph implementation- https://github.com/SoftwareAG/cumulocity-community-plugins/pull/29

n/a

## Proposed changes

Feature allowing to add events and alarms to Data points graph 2.0 config, similiar to old Data points graph implementation (see screen below)
image

New alarm/event selector should be build in similiar manner as Datapoints selector- it should use the same 3-column layout  and give suggestions based on already existing alarms/events:
image

Short demo of new implementation

https://github.com/SoftwareAG/cumulocity-community-plugins/assets/103751383/2168d3d7-d467-4780-9d32-1aa69a5c9a29

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/SoftwareAG/cumulocity-community-plugins/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/SoftwareAG/cumulocity-community-plugins/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/datapoints-graph/alarm-event-selector/alarm-event-selector.component.html [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  </div>
+  <!-- center column -->
+  <div class="inner-scroll bg-component">
+    <ng-template #noDeviceEmptyState>
+      <div class="p-16">
+        <div class="c8y-empty-state text-left">
+          <h1 class="c8y-icon c8y-icon-data-points c8y-icon-duocolor"></h1>
+          <div>
+            <p>
+              <strong translate>{{ timelineTypeTexts.emptyStateTitle }}</strong>
```
Comment (eniosultan) [inline src/datapoints-graph/alarm-event-selector/alarm-event-selector.component.html:38]:
```suggestion
              <strong>{{ timelineTypeTexts.emptyStateTitle | translate }}</strong>
```

File: src/datapoints-graph/alarm-event-selector/alarm-event-selector.component.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  filterString = '';
+  maxNumberOfItems = 50;
+
+  loadingItems = false;
+  assetSelection = new BehaviorSubject<IIdentified>(null);
+  items$: Observable<AlarmOrEvent[]>;
+  filteredItems$: Observable<AlarmOrEvent[]>;
+  filterStringChanges$: Observable<string>;
+  timelineTypeTexts: TimelineTypeTexts;
+  private filterString$ = new BehaviorSubject('');
```
Comment (eniosultan) [inline src/datapoints-graph/alarm-event-selector/alarm-event-selector.component.ts:55]: #minor maybe we can swap the places with the one below :D

File: src/datapoints-graph/alarm-event-selector/pipes/includes-alarm.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import {
+  AlarmDetails,
+  AlarmOrEvent,
```
Comment (eniosultan) [inline src/datapoints-graph/alarm-event-selector/pipes/includes-alarm.pipe.ts:4]: unused import

File: src/datapoints-graph/alarm-event-selector/pipes/includes-alarm.pipe.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  EventDetails,
+} from '../alarm-event-selector.model';
+
+@Pipe({
+  name: 'includesAlarmOrEvent',
+})
+export class IncludesAlarmOrEventPipe implements PipeTransform {
+  transform(itemList: EventDetails[], item?: EventDetails): boolean;
+  transform(itemList: AlarmDetails[], item?: AlarmDetails): boolean;
+  transform(itemList: any[], item?: any): boolean {
```
Comment (eniosultan) [inline src/datapoints-graph/alarm-event-selector/pipes/includes-alarm.pipe.ts:14]: Any thoughts on this approach?
```suggestion
  transform<T extends AlarmDetails | EventDetails>(itemList: T[], item?: T): boolean {
```

**Review by DawidSAG — APPROVED** [review]

**Review by eniosultan — APPROVED** [review]

**Pull request merged**